Makefile

```diff
@@ -69,55 +78,70 @@ publish: build lint
 
 .PHONY: clean
 clean: ## Remove intermediate build artifacts & cache.
-	@rm -rf $(BUILDDIR)/*
-	@rm -rf vendor
+	-rm -r $(BUILDDIR)
+	-rm -r vendor
+	-rm $(VERSIONFILE)
 
 .PHONY: test
 test: ## Run unit tests.
+test: buildbox
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
 		$(BUILDBOX) \
 		dumb-init go test -cover -race -v ./infra/... ./lib/config/...
 
 .PHONY: lint
 lint: ## Run static analysis against source code.
-lint: buildbox
+lint: vendor buildbox
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
-		$(BUILDBOX) dumb-init golangci-lint run \
-		--skip-dirs=vendor \
-		--timeout=2m
+		$(BUILDBOX) dumb-init golangci-lint run
+
+.PHONY: vendor
+vendor: ## Download dependencies into vendor directory.
+vendor: vendor/modules.txt
+
+vendor/modules.txt: go.mod | $(VERSIONFILE) $(BUILDBOX_IIDFILE)
+	docker run $(DOCKERFLAGS) $(BUILDBOX) \
+		dumb-init go mod vendor
+	@touch vendor/modules.txt
```

Comment:
I am not sure I followed why do you need touch here? Isn't `go mod vendor` sufficient?
And overall vendoring should happen during PR, not during the lint or I missed the intentions for vendoring here?

Reply:
Good questions.

> I am not sure I followed why do you need touch here? Isn't go mod vendor sufficient?

No, because if `go mod` doesn't change anything, then `vendor/modules.txt` mtime won't change -- leading make to believe this is out of date when it is actually current.  This is important when version control updates go.mod's mtime without changing content (e.g. checkout to v2.0.0 and then back to master).  In that case, without the touch, this target would remain out of date (go.mod more current than vendor/modules.txt) and re-trigger it and all its downstreams every run.

> And overall vendoring should happen during PR, not during the lint or I missed the intentions for vendoring here?

Having lint depend on vendoring is a nicety for CI and first time users.  Without that dependency the I was seeing the following when lint was run for the first time:

```
docker run --rm=true -u 1000:1000 -v /drone/src:/go/src/github.com/gravitational/robotest -v /drone/src/build:/go/src/github.com/gravitational/robotest/build -w /go/src/github.com/gravitational/robotest \
    --env="GO111MODULE=off" \
    robotest:buildbox dumb-init golangci-lint run \
    --skip-dirs=vendor \
    --timeout=2m
level=warning msg="[runner] Can't run linter goanalysis_metalinter: S1009: failed prerequisites: [(inspect@github.com/gravitational/robotest/lib/loc, isgenerated@github.com/gravitational/robotest/lib/loc): analysis skipped: errors in package: [/go/src/github.com/gravitational/robotest/lib/loc/loc.go:24:2: could not import github.com/gravitational/trace (lib/loc/loc.go:24:2: cannot find package \"github.com/gravitational/trace\" in any of:\n\t/go/src/vendor/github.com/gravitational/trace (vendor tree)\n\t/go/src/github.com/gravitational/trace (from $GOROOT)\n\t/gopath/src/github.com/gravitational/trace (from $GOPATH))]]"
level=warning msg="[runner] Can't run linter unused: buildir: failed to load package : could not load export data: no export data for \"github.com/gravitational/trace\""
level=error msg="Running error: buildir: failed to load package : could not load export data: no export data for \"github.com/gravitational/trace\
```

For instance: https://drone.gravitational.io/gravitational/robotest/17/1/4

This is because robotest doesn't include `vendor/` in the git repo, unlike our other repositories.